plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsMetadata.java

```diff
@@ -145,4 +159,22 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     {
         return ((SheetsColumnHandle) columnHandle).getColumnMetadata();
     }
+
+    @Override
+    public Optional<TableFunctionApplicationResult<ConnectorTableHandle>> applyTableFunction(ConnectorSession session, ConnectorTableFunctionHandle handle)
+    {
+        if (!(handle instanceof Query.QueryFunctionHandle)) {
+            return Optional.empty();
+        }
+
+        ConnectorTableHandle tableHandle = ((Query.QueryFunctionHandle) handle).getTableHandle();
+        ConnectorTableSchema tableSchema = getTableSchema(session, tableHandle);
+        Map<String, ColumnHandle> columnHandlesByName = getColumnHandles(session, tableHandle);
+        List<ColumnHandle> columnHandles = tableSchema.getColumns().stream()
+                .map(ColumnSchema::getName)
+                .map(columnHandlesByName::get)
+                .collect(toImmutableList());
+
+        return Optional.of(new TableFunctionApplicationResult<>(tableHandle, columnHandles));
```

Comment:
Please revert the if condition.

Reply:
I copied this pattern from `applyTableFunction` in other connectors, see bigquery and base-jdbc:
https://github.com/trinodb/trino/blob/master/plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java#L701-L704
https://github.com/trinodb/trino/blob/master/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java#L606-L608

Reply:
I think we'll eventually revert as we add more Table Functions. You can leave it for now.